app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,105 @@
+class OrdersController < ApplicationController
+  
+  before_action :authenticate_user!
+  before_action :set_journey, only: %i(new create checkout)
+  before_action :set_order, only: %i(checkout charge)
+  before_action :instantiate_order, only: %i(create)
+  before_action :ensure_booking_is_active, only: %i(new checkout create)
+  
+  def checkout
+  end
+  
+  def new
+    @minutes = Order.calculate_minutes(@journey)
+    @order = Order.new
+    authorize(@order)
+  end 
+  
+  def create
+    @order.calculate_consumer_total_cents
```

Comment:
Yes, mutating instance variables is a side-effect. `!` informs us of side effects